base/linalg/dense.jl

```diff
@@ -18,6 +20,8 @@ function scale!{T<:BlasFloat}(X::Array{T}, s::T)
     X
 end
 
+scale!{T<:BlasFloat}(s::T, X::Array{T}) = scale!(X, s)
```

Comment:
Added to #16772. Being explicit with 3 arguments seems like a much more predictable API when you're writing generic code that has to deal with various combinations of input types.